packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -143,10 +143,10 @@ const TablePagination = React.forwardRef(function TablePagination(props, ref) {
             labelId={labelId}
             {...SelectProps}
           >
-            {rowsPerPageOptions.map((rowsPerPageOption) => (
+            {rowsPerPageOptions.map((rowsPerPageOption, index) => (
               <MenuItemComponent
                 className={classes.menuItem}
-                key={rowsPerPageOption.value ? rowsPerPageOption.value : rowsPerPageOption}
+                key={index}
```

Comment:
Given the test case, we could just leverage the label. Though I don't understand why you would duplicate the value to begin with. Since the order can change, using the index actually is problematic.

Reply:
You would need to discuss this point with @oliviertassinari , he was the one who added the change to `index`. I would be fine with either approach, since both have their advantages and disadvantages, as I already explained here: https://github.com/mui-org/material-ui/pull/25309#issuecomment-799273288

Reply:
>  I would be fine with either approach, since both have their advantages and disadvantages

What are the disadvantages of using `label`?

Reply:
See https://github.com/mui-org/material-ui/pull/25309#pullrequestreview-611588082, as I understood @oliviertassinari 's point is that using `index` would guarantee a unique key here, which would be a more pragmatic approach, since it would **ultimately** solve **this** issue. If we change it to something else (like `label`) instead, we  run the risk of missing another aspect that could result in another PR in the future attempting to fix the issue, considering there has already been a PR in the past which attempted to solve this issue.

But of course, as you and I (https://github.com/mui-org/material-ui/pull/25309#issuecomment-799273288) pointed out, this also has disadvantages. As I mentioned, since there are advantages and disadvantages to both approaches, personally I would be fine with either solution.

Reply:
>  since it would ultimately solve this issue.

I understand that this would fix your issue. As library maintainers it is our job to factor in everybody into the best possible implementation. When a trade-off has to be made between what certain devs "want" (in this case: a constructed hypothetical) vs what the end-user might experience we prefer the end-user. This is standard practice on the web and I hope that we can work together towards this goal.